File: src/main/kotlin/com/indefiniteStudies/model/entities/Comment.kt
package com.indefiniteStudies.model.entities

import org.jetbrains.exposed.sql.Column
import org.jetbrains.exposed.sql.Table
import org.jetbrains.exposed.sql.javatime.datetime
import java.time.LocalDateTime

object Comment : Table("notes") {
    val id: Column<Int> = integer("id").autoIncrement()
    val text: Column<String> = text("text")
    val userId: Column<Int> = integer("user_id")
    val noteId: Column<Int> = integer("note_id")
    val linkedCommentId: Column<Int> = integer("linked_comment_id")
    val state: Column<String> = varchar("state", 256)
    val createDate: Column<LocalDateTime> = datetime("create_date")
    val lastUpdateDate: Column<LocalDateTime> = datetime("last_update_date")
}

File: internal/db/entities/comment.go
package entities

import "time"

type Comment struct {
	Id               int
	Text             string
	UserId           int
	NoteId           int
	LinkdedCommentId int
	State            string
	CreateDate       time.Time
	LastUpdateDate   time.Time
}

const (
	COMMENT_STATE_NEW     string = "NEW"
	COMMENT_STATE_BLOCKED string = "BLOCKED"
	COMMENT_STATE_DELETED string = "DELETED"
)
